test/test_optim.py

```diff
@@ -567,11 +536,12 @@ def test_old_pattern_warning(self):
         epochs = 35
         with warnings.catch_warnings(record=True) as ws:
             warnings.simplefilter("always")  # allow any warning to be raised
+            warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
Is the warning filtering here new? Why are these tests raising deprecation warnings? Shouldn't our test code be using the non-deprecated codepath?

Reply:
Thanks for catching this. I'll remove them.